src/streaming/ManifestLoader.js

```diff
@@ -194,6 +194,16 @@ function ManifestLoader(config) {
                         logger.debug('BaseURI set by Location to: ' + baseUri);
                     }
 
+                    // If there is a mismatch between the manifest's specified duration and the total duration of all periods,
+                    // overwrite the manifest's duration attribute. This is a patch for if a manifest is generated incorrectly.
+                    if (manifest.mediaPresentationDuration && manifest.Period.length > 1) {
```

Comment:
Please add a settings flag to enable this workaround. It can be enabled by default. In case we see any side effects we did not think of before users should be able to disable this workaround.

It can be something like `streaming.enableManifestDurationMismatchFix : true` . Or any other descriptive name you can think of.